client/src/components/WinLoss.svelte

```diff
@@ -0,0 +1,163 @@
+<script lang="ts">
+  import { authStore as user } from "$stores/auth";
+  import type { PrivatePlayer, Lobby } from "../../../functions/src/firestore-types/lobby";
+  import { onMount } from "svelte";
+  import { getPrivatePlayer, lobbyReturn } from "$lib/firebase/winloss";
+
+  export let lobbyCode: string;
+  export let privatePlayer: PrivatePlayer | undefined;
+  export let lobby: Lobby | undefined;
+  let lobbyRoles: PrivatePlayer | undefined;
+
+  let end: "Cat Win" | "Cat Lose" | "Catfish Lose" | "Catfish Win" | null = null;
+  let catfishDisplayname: string[] = []; // catfish's dislplay name used in the html lobbyCode
+  let catfishList: string = "";
+  let catfishCounter = 0;
+
+  onMount(async () => {
+    let aliveCatCount = 0;
+    // function to count the number of players currently alive
+    if (lobby !== undefined) {
+      for (let i = 0; i < lobby.uids.length; i++) {
+        if (lobby.players[i].alive == true) {
+          // count the number of remaining players who are alive
+          lobbyRoles = await getPrivatePlayer(lobbyCode, lobby.uids[i]);
+          if (lobbyRoles !== undefined) {
+            if (lobbyRoles.role == "CAT") {
+              aliveCatCount++;
+              // within the alive players, count the number that are cats
+            } else {
+              catfishDisplayname[catfishCounter] = lobby.players[i].displayName; // create a list of catfishes
+              catfishCounter++;
```

Comment:
did you know about `push`?
what you did works but push will make your life easier if you didn't already know about it.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push